services/coverity/coverity-scan.tester.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+
+const Joi = require('joi')
+const t = (module.exports = require('../create-service-tester')())
+
+t.create('live: known project id')
+  .get('/3997.json')
+  .expectJSONTypes(
+    Joi.object().keys({
+      name: 'coverity',
+      value: Joi.string().regex(/passing|passed .* new defects|pending|failed/),
+    })
+  )
+
+t.create('live: unknown project id')
+  .get('/abc.json')
+  // Coverity actually returns an HTTP 200 status with an HTML page when the project is not found.
+  .expectJSON({ name: 'coverity', value: 'unparseable json response' })
+
+t.create('404 response')
+  .get('/1.json')
+  .intercept(nock =>
+    nock('https://scan.coverity.com/projects/1')
+      .get('/badge.json')
+      .reply(404)
+  )
+  .expectJSON({ name: 'coverity', value: 'project not found' })
+
+t.create('passed')
+  .get('/2.json')
+  .intercept(nock =>
+    nock('https://scan.coverity.com/projects/2')
+      .get('/badge.json')
+      .reply(200, {
+        message: 'passed',
+      })
+  )
+  .expectJSON({ name: 'coverity', value: 'passing' })
```

Comment:
If you want to mock a response for each case, we could usefully make the calls with the `?style=_shields_test` param here and test the colouring logic too. There's an example of this in the docs https://github.com/badges/shields/blob/master/doc/service-tests.md#5-mocking-responses

Reply:
Excellent! I was wondering about color validation. Will add this (I should probably update some other service tests to include this as well)

Reply:
Another approach which can be taken here is a Sazerac-based test of `render` in a `.spec.js` file.

Reply:
Is there a preferred approach or is either good?